packages/frontend/editor-ui/src/features/chatHub/chat.store.ts

```diff
@@ -349,6 +351,31 @@ export const useChatStore = defineStore(CHAT_STORE, () => {
 			alternatives: [],
 		});
 
+		if (!model || !credentials) {
+			addMessage(sessionId, {
+				id: replyId,
```

Comment:
Doesn't this mess up the `previousMessageId` in followup messages, as this doesn't exist on the DB?

Reply:
also it's a bit messy as in that case the response won't be a streamed one but we set the streaming headers. this needs work

Reply:
There are different error modes:

1. Submit without meeting preconditions (the code above is trying to catch)
2. General API call failure (incl. network error)
3. Error in workflow execution

On second thought, I think it's better to handle 1 and 2 solely in FE even if we wanna show them like a response from the agent. Can we leave it like this and improve error handling in general in a separate PR?